### PR TITLE
v0.0.14

### PR DESCRIPTION
**Pull request by smalltown**
## Elastikube
1. Master work group adopt AWS launch template
2. Master apiserver lb can choose public/private
3. Change the folder structure for Terratest
4. Add a test for example elastikube-cluster

### CODE REVIEW

File: test/elastikube_cluster_test.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+}
+
+func createKubeconfig(t *testing.T, source string, kubeconfig string) (string, error) {
+	kubeconfigPath := fmt.Sprintf("%s/.test-data/kubeconfig", source)
+
+	f, err := os.Create(kubeconfigPath)
+	if err != nil {
+		return "", err
+	}
+	l, err := f.WriteString(kubeconfig)
```
Comment (owensengoku) [inline test/elastikube_cluster_test.go]: Could use ` defer f.Close()` here

File: test/elastikube_cluster_test.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+		return "", nil
+	})
+}
+
+func testKubernetes(t *testing.T, kubectlOptions *k8s.KubectlOptions, exampleFolder string) {
+
+	// It can take several minutes for the Kubernetes cluster to boot up, so retry a few times
+	maxRetries := 30
+	timeBetweenRetries := 30 * time.Second
+	description := fmt.Sprint("Access Kubernetes cluster")
```
Comment (owensengoku) [inline test/elastikube_cluster_test.go]: what vairable for `fmt.Sprint`?

File: test/elastikube_cluster_test.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+	bastionHost := &ssh.Host{
+		Hostname:         bastionPublicIP,
+		SshUserName:      "ubuntu",
+		OverrideSshAgent: sshAgent,
+	}
+
+	// It can take a minute or so for the Instance to boot up, so retry a few times
+	maxRetries := 30
+	timeBetweenRetries := 5 * time.Second
+	description := fmt.Sprintf("SSH with Agent to bastion host %s", bastionPublicIP)
```
Comment (owensengoku) [inline test/elastikube_cluster_test.go]: Could consider use 

```


const (
	MaxRetries         = 30
	TimeBetweenRetries = 5 * time.Second
)

```
at the top of this file
Ref: https://play.golang.org/p/Pd-cbrXDPkQ

File: test/elastikube_cluster_test.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+	kubeTunnel := k8s.NewTunnel(kubectlOptions, k8s.ResourceTypeService, testServiceName, testServiceLocal, testServiceRemote)
+	kubeTunnel.ForwardPort(t)
+	defer kubeTunnel.Close()
+
+	// Test the endpoint for up to 5 minutes. This will only fail if we timeout waiting for the service to return a 200
+	// response.
+	http_helper.HttpGetWithRetryWithCustomValidation(
+		t,
+		fmt.Sprintf("http://%s", kubeTunnel.Endpoint()),
+		30,
```
Comment (owensengoku) [inline test/elastikube_cluster_test.go]: Could replase `30` with `MaxRetries`

**Comment by smalltown** [comment]
@owensengoku thank for your reviewing and feedback very much, after testing, I try to merge this PR

**Pull request merged**